frontend/pages/policies/ManagePoliciesPage/ManagePoliciesPage.tsx

```diff
@@ -311,117 +274,63 @@ const ManagePolicyPage = (managePoliciesPageProps: {
-    // that would be triggered on a change to selectedTeamId.
-    if (currentTeam) {
-      setSelectedTeamId(currentTeam.id);
-    } else {
-      teamId !== null && setSelectedTeamId(teamId);
-    }
-  }, [isOnGlobalTeam, location, userTeams]);
+  const inheritedPoliciesButtonText = (
+    showPolicies: boolean,
+    count: number
+  ) => {
+    return `${showPolicies ? "Hide" : "Show"} ${count} inherited ${
+      count > 1 ? "policies" : "policy"
+    }`;
+  };
 
-  // Watch for selected team changes and call getPolicies to make new policies API request.
-  useEffect(() => {
-    // Null case must be distinguished from 0 (which is used as the id for the "All teams" option)
-    // so a falsiness check cannot be used here. Null case here allows us to skip API call.
-    if (selectedTeamId !== null) {
-      if (isOnGlobalTeam || isAnyTeamMaintainerOrTeamAdmin) {
-        getGlobalPolicies();
-      }
-      if (selectedTeamId) {
-        getTeamPolicies(selectedTeamId);
-      }
-    }
-    getFailingPoliciesWebhook();
-  }, [
-    getGlobalPolicies,
-    getTeamPolicies,
-    getFailingPoliciesWebhook,
-    isAnyTeamMaintainerOrTeamAdmin,
-    isOnGlobalTeam,
-    selectedTeamId,
-  ]);
+  const canAddOrRemovePolicy =
+    isGlobalAdmin || isGlobalMaintainer || isTeamMaintainer || isTeamAdmin;
 
-  // Pull osquery policy update interval value from config, reformat, and set as updateInterval.
-  useEffect(() => {
-    if (config) {
-      const { osquery_policy: interval } = config;
-      interval &&
-        setUpdateInterval(secondsToHms(inMilliseconds(interval) / 1000));
-    }
-  }, [config]);
+  const policyUpdateInterval =
+    secondsToHms(inMilliseconds(config?.osquery_policy || 0) / 1000) ||
+    "osquery policy update interval";
 
-  // If the user is free tier or if there is no selected team, we show the default description.
-  // We also want to check selectTeamId for the null case so that we don't render the element prematurely.
-  const showDefaultDescription =
-    isFreeTier || (isPremiumTier && !selectedTeamId && selectedTeamId !== null);
+  const showTeamDescription = isPremiumTier && !!teamId;
 
   const showInfoBanner =
-    (selectedTeamId && !isTeamPoliciesError && !!teamPolicies?.length) ||
-    (!selectedTeamId &&
-      selectedTeamId !== null &&
-      !isGlobalPoliciesError &&
-      !!globalPolicies?.length);
+    (teamId && !teamPoliciesError && !!teamPolicies?.length) ||
+    (!teamId && !globalPoliciesError && !!globalPolicies?.length);
 
   const showInheritedPoliciesButton =
-    !!selectedTeamId && !!globalPolicies?.length && !isGlobalPoliciesError;
+    !!teamId &&
+    !isLoadingTeamPolicies &&
+    !teamPoliciesError &&
+    !isLoadingGlobalPolicies &&
+    !globalPoliciesError &&
+    !!globalPolicies?.length;
+
+  // Validate team_id from URL query params and redirect invalid cases to default policy page
+  useEffect(() => {
+    if (availableTeams !== null && availableTeams !== undefined) {
```

Comment:
Would a simple `if (availableTeams)` work here? Or do we need to explicitly check for `null` and `undefined`?